plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1011,15 +1012,216 @@ public void testPathColumn()
             assertUpdate("INSERT INTO " + table.getName() + " SELECT 'second'", 1);
             String secondFilePath = (String) computeScalar("SELECT \"$path\" FROM " + table.getName() + " WHERE x = 'second'");
 
-            // Verify predicate correctness on $path column
-            assertQuery("SELECT x FROM " + table.getName() + " WHERE \"$path\" = '" + firstFilePath + "'", "VALUES 'first'");
-            assertQuery("SELECT x FROM " + table.getName() + " WHERE \"$path\" <> '" + firstFilePath + "'", "VALUES 'second'");
-            assertQuery("SELECT x FROM " + table.getName() + " WHERE \"$path\" IN ('" + firstFilePath + "', '" + secondFilePath + "')", "VALUES ('first'), ('second')");
-            assertQuery("SELECT x FROM " + table.getName() + " WHERE \"$path\" IS NOT NULL", "VALUES ('first'), ('second')");
+            // Verify predicate correctness on $path column, and check it is pusheddown
+            assertThat(query("SELECT x FROM " + table.getName() + " WHERE \"$path\" = '" + firstFilePath + "'"))
+                    .matches("VALUES CAST('first' AS VARCHAR)")
+                    .isFullyPushedDown();
+            assertThat(query("SELECT x FROM " + table.getName() + " WHERE \"$path\" <> '" + firstFilePath + "'"))
+                    .matches("VALUES CAST('second' AS VARCHAR)")
+                    .isFullyPushedDown();
+            assertThat(query("SELECT x FROM " + table.getName() + " WHERE \"$path\" IN ('" + firstFilePath + "', '" + secondFilePath + "')"))
+                    .matches("VALUES (CAST('first' AS VARCHAR)), (CAST('second' AS VARCHAR))")
+                    .isFullyPushedDown();
+            assertThat(query("SELECT x FROM " + table.getName() + " WHERE \"$path\" IS NOT NULL"))
+                    .matches("VALUES (CAST('first' AS VARCHAR)), (CAST('second' AS VARCHAR))")
+                    .isFullyPushedDown();
             assertQueryReturnsEmptyResult("SELECT x FROM " + table.getName() + " WHERE \"$path\" IS NULL");
         }
     }
 
+    @Test
+    public void testOptimizeWithPathColumn()
```

Comment:
Could you confirm if we have a test with special characters on `$path` column? 
The `$path` in the following SELECT statement contains `b=a%23sharp`:
```sql
CREATE TABLE test WITH (partitioned_by = array['b']) AS SELECT 1 a, 'a#sharp' b;
SELECT "$path" FROM test;
```

We should verify that a condition with `...b=a%23sharp...` matches and `...b=a#sharp...` doesn't match.

Reply:
Extend the test `testPathColumn`